Labs4

New1sens opened the pull request.

New1sens committed:
dz

New1sens committed:
Delete .idea directory

integralpir commented on Второй семестр/Labs4/hes_table.java:
```diff
@@ -0,0 +1,149 @@
+    private final float loadFactor; // Коэффициент загрузки
+
+    // Вложенный класс для хранения элементов таблицы
+    private static class Entry {
+        final int key; // Ключ элемента (неизменяемый)
+        int value; // Значение элемента
+        Entry next; // Ссылка на следующий элемент в цепочке (для разрешения коллизий)
+
+
+        Entry(int key, int value, Entry next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
```
Это очень грязный код, вложенные классы должны лежать в самом низу основного класса, а лучше вообще в другом файле

integralpir commented on Второй семестр/Labs4/hes_table.java:
```diff
@@ -0,0 +1,149 @@
+        this.loadFactor = loadFactor; // Установка коэффициента загрузки
+        this.table = new Entry[initialCapacity]; // Создание массива бакетов
+        this.threshold = (int)(initialCapacity * loadFactor);
+    }
```
В C# есть возможность устанавливать дефолтные значения параметров метода, не вижу смысла в перегрузке конструктора

integralpir commented on Второй семестр/Labs4/hes_table.java:
```diff
@@ -0,0 +1,149 @@
+
+    // Метод для добавления или обновления элемента
+    public void put(int key, int value) {
+                if (size >= threshold) {
```
Отступ некрасивый

New1sens replied:
Исправил работу.

integralpir reviewed (changes requested):
По логике все неплохо, но в коде надо прибраться. Устрани замечания и лабу зачту

New1sens committed:
Update hes_table.java

Исправил все недочеты, извиняюсь что так поздно, только сейчас выдалась возможность сесть за компьютер
